frontend/packages/console-shared/src/utils/firehose.ts

```diff
@@ -0,0 +1,39 @@
+import { FirehoseResult } from '@console/internal/components/utils';
+import { K8sKind, K8sResourceCommon, K8sResourceKind } from '@console/internal/module/k8s';
+import * as _ from 'lodash';
+
+export const isLoaded = <A extends K8sResourceCommon | K8sResourceCommon[] = K8sResourceKind[]>(
+  result: FirehoseResult<A>,
+) => result && result.loaded;
+
+export const getLoadedData = <
```

Comment:
This is actually the only one used in metal3 and you can just do `result?.data`

Reply:
Assuming `result.data` is not set upon an error, I agree with @rawagner - doing `result?.data` should be enough here.

Reply:
get loaded data is more advanced helper which IMO has its use case. We just care about getting the data or `[]` if anything went wrong. But we don't want to care about those small things like `loadError` and `loaded` property